include/torch-mlir/Dialect/Torch/IR/TorchOps.td

```diff
@@ -958,6 +958,101 @@ def Torch_ValueTensorLiteralOp : Torch_Op<"vtensor.literal", [
   let hasFolder = 1;
 }
 
+def Torch_ExternalNonValueTensorLiteralOp : Torch_Op<"tensor.external.literal", [
+    AllowsTypeRefinement,
+    AllowedInModuleInitializer
+  ]> {
+  let summary = "Create a value of !torch.tensor type from an external literal";
+  let description = [{
+    Example:
+    ```
+    %0 = torch.tensor.external.literal(@fc.weight) : !torch.tensor<[2,3],f32>
+    ```
+
+    This is the non-value-semantic version of `torch.vtensor.external.literal`.
+    See the documentation there for more context.
+  }];
+  let arguments = (ins FlatSymbolRefAttr:$sym_name);
+  let results = (outs Torch_NonValueTensorType:$result);
+
+  let assemblyFormat = [{
+    `(` $sym_name `)` attr-dict `:` qualified(type($result))
+  }];
+}
+
+def Torch_ExternalValueTensorLiteralOp : Torch_Op<"vtensor.external.literal", [
+    AllowsTypeRefinement,
+  ]> {
+  let summary = "Create a value of !torch.vtensor type from an external literal";
+  let description = [{
+    Example:
+    ```
+    %0 = torch.vtensor.external.literal(@fc.weight) : !torch.vtensor<[2,3],f32>
+    ```
+
+    The use of the term "literal" for an external reference is a bit strange,
+    but is a natural byproduct of Torch-MLIR's evolution. Currently, Torch-MLIR
+    is in the process of aligning itself with PyTorch's vision of a trace-based
+    future, where backend compilers (here meaning Torch-MLIR's main lowering
+    pipeline to the backend contract) only handle traces, rather than
+    full stateful programs. For in-process on-demand compilation, such as a
+    TorchDynamo backend compiler, the traces never see any program state.
+    For Ahead-of-Time compilation, this gets trickier. To handle
+    programs for Ahead-of-Time, the idea (already proved out in the JAX space,
+    and in progress in PyTorch) is to capture traces and then use an outer layer
+    with a more general programming model (e.g. stateful,
+    more "Python-like"/dynamic) to orchestrate calls into the
+    traces and manage parameters and buffers. This outer layer would be manually
+    programmed by users, which avoids ~intractable problems related to lowering
+    fully general TorchScript modules which simultaneously have the bulk tensor
+    compute and orchestrate logic all in one program. Although this seems like
+    a usability reduction, it actually results in a much less brittle compiler,
```

Comment:
Agreed, but I think since we qualify it as in-progress it is fine. Also again this is really completely specific to this one op so I think it is fine to keep it here. It is sort of an elaboration of the context in the long-term roadmap doc in a sense.